libs/sqf/src/analyze/lints/s20_bool_static_comparison.rs

```diff
@@ -0,0 +1,162 @@
+use std::{ops::Range, sync::Arc};
+
+use hemtt_common::config::LintConfig;
+use hemtt_workspace::{lint::{AnyLintRunner, Lint, LintRunner}, reporting::{Code, Codes, Diagnostic, Processed, Severity}};
+
+use crate::{analyze::SqfLintData, BinaryCommand, Expression};
+
+crate::lint!(LintS20BoolStaticComparison);
+
+impl Lint<SqfLintData> for LintS20BoolStaticComparison {
+    fn ident(&self) -> &str {
+        "bool_static_comparison"
+    }
+
+    fn sort(&self) -> u32 {
+        200
+    }
+
+    fn description(&self) -> &str {
+        "Checks for a variable being compared to `true` or `false`"
+    }
+
+    fn documentation(&self) -> &str {
+        r"### Example
+
+**Incorrect**
+```sqf
+if (_x == true) then {};
+if (_y == false) then {};
+```
+**Correct**
+```sqf
+if (_x) then {};
+if (!_y) then {};
+```
+"
+    }
+
+    fn default_config(&self) -> LintConfig {
+        LintConfig::help()
+    }
+
+    fn runners(&self) -> Vec<Box<dyn AnyLintRunner<SqfLintData>>> {
+        vec![Box::new(Runner)]
+    }
+}
+
+struct Runner;
+impl LintRunner<SqfLintData> for Runner {
+    type Target = crate::Expression;
+
+    fn run(
+        &self,
+        _project: Option<&hemtt_common::config::ProjectConfig>,
+        config: &LintConfig,
+        processed: Option<&hemtt_workspace::reporting::Processed>,
+        target: &Self::Target,
+        _data: &SqfLintData,
+    ) -> Codes {
+        let Some(processed) = processed else {
+            return Vec::new();
+        };
+        let Expression::BinaryCommand(BinaryCommand::Eq | BinaryCommand::NotEq, lhs, rhs, _) = target else {
+            return Vec::new();
+        };
+        
```

Comment:
something like this if you want to match `isNotEqualTo` as well?
```suggestion
        if cmd.as_str() != "==" && cmd.as_str() != "!=" && cmd.as_str() != "isEqualTo" && cmd.as_str() != "isNotEqualTo" {
            return Vec::new();
        }
```

Reply:
This creates a bunch of false positives
![image](https://github.com/user-attachments/assets/17474cb2-d130-464e-a6c5-83520e5da8a3)